k9mail/src/main/java/com/fsck/k9/activity/compose/AttachmentPresenter.java

```diff
@@ -175,6 +185,27 @@ public void processMessageToForward(MessageViewInfo messageViewInfo) {
         }
     }
 
+    public void processMessageToForwardAsAttachment(MessageViewInfo messageViewInfo) throws IOException, MessagingException {
+        boolean isMissingParts = !allPartsAvailable(messageViewInfo, false);
+        if (isMissingParts) {
+            attachmentMvpView.showMissingAttachmentsPartialMessageForwardWarning();
+        } else {
+            File tempFile = File.createTempFile("pre", ".tmp");
+            tempFile.deleteOnExit();
```

Comment:
Huh, you're right `AttachmentContentLoader` does this as well. Ugh. We really shouldn't be doing this, but I'm not sure what a good way to deal with this is.. @cketti ?